hedera-node/test-clients/src/itest/java/InProcessIntegrationTestBase.java

```diff
@@ -0,0 +1,237 @@
+import com.swirlds.platform.gossip.chatter.config.ChatterConfig;
+import com.swirlds.platform.gossip.sync.config.SyncConfig;
+import com.swirlds.platform.state.EmergencyRecoveryManager;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import com.swirlds.platform.uptime.UptimeConfig;
+import com.swirlds.virtualmap.config.VirtualMapConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.concurrent.Executors;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.io.TempDir;
+
+/** Base class for integration tests */
+public abstract class InProcessIntegrationTestBase extends TestBase {
+    static {
+        System.setProperty("java.util.logging.manager", "org.apache.logging.log4j.jul.LogManager");
+    }
+
+    static Hedera hedera;
+
+    /** Create and start a node, and configure the HapiSpec to use that node. */
+    @BeforeAll
+    static void beforeAll(@TempDir @NonNull final Path tmpDir) throws Exception {
+        // Setup logging
+        try (final var log4j2ConfigFile = InProcessIntegrationTestBase.class.getResourceAsStream("/log4j2.xml")) {
+            final var source = new ConfigurationSource(requireNonNull(log4j2ConfigFile));
+            try (final var ignored = Configurator.initialize(null, source)) {
+                // Nothing to do here.
+            }
+        }
+
+        // Setup the constructable registry so the platform can deserialize the state.
+        final var registry = ConstructableRegistry.getInstance();
+        registry.reset();
+        registry.registerConstructables("com.swirlds.merklemap");
+        registry.registerConstructables("com.swirlds.jasperdb");
+        registry.registerConstructables("com.swirlds.fcqueue");
+        registry.registerConstructables("com.swirlds.virtualmap");
+        registry.registerConstructables("com.swirlds.common.merkle");
+        registry.registerConstructables("com.swirlds.common");
+        registry.registerConstructables("com.swirlds.merkle");
+        registry.registerConstructables("com.swirlds.merkle.tree");
+
+        // 1. Create a configuration instance with any desired overrides.
+        final var factory = ServiceLoader.load(ConfigurationBuilderFactory.class);
+        final var configBuilder = factory.findFirst().orElseThrow().create();
+        final var config = configBuilder
+                //                .withSource(mappedSettingsConfigSource)
+                //                .withSource(configPropertiesConfigSource)
+                //                .withSource(threadCountPropertyConfigSource)
```

Comment:
Intentionally commented out?

Reply:
Yes, it is a work in progress. But I can remove these.